examples/lock-app/telink/include/BoltLockManager.h

```diff
@@ -99,8 +106,16 @@ class BoltLockManager
     UserData mUserData[CONFIG_LOCK_NUM_USERS];
     EmberAfPluginDoorLockUserInfo mUsers[CONFIG_LOCK_NUM_USERS] = {};
 
-    CredentialData mCredentialData[CONFIG_LOCK_NUM_CREDENTIALS];
-    EmberAfPluginDoorLockCredentialInfo mCredentials[CONFIG_LOCK_NUM_CREDENTIALS] = {};
+    struct LockCredentialInfo
+    {
+        DlCredentialStatus status;
+        CredentialTypeEnum credentialType;
+        chip::FabricIndex createdBy;
+        chip::FabricIndex modifiedBy;
+        uint8_t credentialData[CONFIG_LOCK_CREDENTIAL_INFO_MAX_TYPES];
+        size_t credentialDataSize;
+    };
+    std::vector<std::vector<LockCredentialInfo>> mCredentials;
```

Comment:
So what gives this a nonzero capacity?  There are various capacity tests in the added code, as far as I can tell....

Is this app able to pass https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/cluster/door_lock_Cluster.adoc#tc-drlk-2-11-verification-for-multiple-credential-types-dut-server for whatever feature map it has?

Reply:
Thanks, `mCredentials` is allocated in constructor for now. Actually we have not run any certification tests yet as we don't have plans for certification in nearest time. So these apps have quite a short functionality(demo) for the present